source/Nevermore/Mapping/ReferenceCollectionReaderWriter.cs

```diff
@@ -15,20 +16,28 @@ public override object Read(object target)
             if (value == null || value.Count == 0)
                 return "";
 
-            return $"|{string.Join("|", value)}| ";
+            return UnParse(value);
         }
 
         public override void Write(object target, object value)
         {
-            var valueAsString = (value ?? string.Empty).ToString().Split('|');
-
             var collection = base.Read(target) as ReferenceCollection;
             if (collection == null)
             {
                 base.Write(target, collection = new ReferenceCollection());
             }
 
-            collection.ReplaceAll(valueAsString.Where(item => !string.IsNullOrWhiteSpace(item)));
+            collection.ReplaceAll(Parse(value?.ToString()));
+        }
+
+        public static IEnumerable<string> Parse(string value)
```

Comment:
if you'd prefer. I like having it close to where its used unless it makes sense to seperate rather than for the sake of separation.